RNA-seq/02-gastric_cancer_tximeta.Rmd

```diff
@@ -0,0 +1,213 @@
+                      names = sample_names)
+```
+
+We have more information about these samples stored in the metadata file that we will also want stored in `coldata`. 
+Let's read in the sample metadata from the TSV file.
+
+```{r sample_meta_df, live = TRUE}
+# Read in the sample metadata TSV file and have a look
+sample_meta_df <- readr::read_tsv(meta_file)
+sample_meta_df
+```
+
+We'll want this information to be added to the `coldata`. 
+
+```{r join-sample_meta_df}
+coldata <- coldata %>%
+  dplyr::inner_join(sample_meta_df, by = c("names" = "srr_accession"))
+
+coldata
+```
+
+## Import expression data with `tximeta`
+
+Using the `coldata` data frame that we set up, we can now run the `tximeta()` to import our expression data while automatically finding and associating the transcript annotations that were used when we performed the quantification.
+
+The first time you run `tximeta()` you will get a message about storing the data in a [cache](https://en.wikipedia.org/wiki/Cache_(computing) so that it will retrieve the data faster the next time.
+We recommend you use the cache.
+
+```{r tximeta, live = TRUE}
+txi_data <- tximeta(coldata)
+```
+
+*tximeta currently works easily for most human and mouse datasets, but requires a [few more steps for other species](https://bioconductor.org/packages/release/bioc/vignettes/tximeta/inst/doc/tximeta.html#What_if_checksum_isn%E2%80%99t_known).
+
+## Summarize to gene
+
+We'll summarize to the gene level using the `summarizeToGene()` function.
+
+```{r summarize-gene}
+# Summarize to the gene level
+gene_summarized <- summarizeToGene(txi_data) 
+```
+
+We can use the `class` function to see what type of object `gene_summarized` is.
+
+```{r class, live = TRUE}
+# Check what type of object `gene_summarized` is
+class(gene_summarized)
+```
+
+This tells us that `gene_summarized` is an object called a [`SummarizedExperiment`](https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/SummarizedExperiment-class) which can be handled by functions from the package of the same name. 
+We more specifically have a `RangedSummarizedExperiment` which is a more specific type of `SummarizedExperiment`.
+
+`SummarizedExperiment` objects have this general structure:  
+
+![SummarizedExperiment](diagrams/SummarizeExperiment-structure.png)
+
+This figure is from this handy vignette about [`SummarizedExperiment` objects](https://www.bioconductor.org/packages/devel/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html).
+
+As shown in the diagram, we can use some of these functions to extract data from our `SummarizedExperiment` object. 
+`rowData()` on our object shows all the gene information that `tximeta` set up!
+
+```{r}
+# rowData() shows us our gene annotation
+rowData(gene_summarized)
+```
+
+The `assay` slot in `SummarizedExperiment`s hold the experiment data, a gene x sample matrix. 
+In this case, it has our gene-level abundance information. 
```

Comment:
Strictly speaking not all assays are gene x sample matrices (lengths is not) so I am adjusting this.
```suggestion
The `assay` slot in `SummarizedExperiment`s holds data from the experiment. 
In this case, it will include our gene-level abundance information stored as a gene x sample matrix. 
```

Reply:
Ah. Yeah. Good point.